src/Microsoft.DocAsCode.App/RunServe.cs

```diff
@@ -43,11 +47,104 @@ public static void Exec(string folder, string host, int? port)
+                using var process = LaunchBrowser(launchUrl);
+
+                // Wait until server exited.
+                app.WaitForShutdown();
+
+                // process object is disposed. (Note:Launched process remain running.after dispose)
+            }
+            else
+            {
+                app.Run();
+            }
         }
         catch (System.Reflection.TargetInvocationException)
         {
             Logger.LogError($"Error serving \"{folder}\" on {url}, check if the port is already being in use.");
         }
     }
+
+    /// <summary>
+    /// Resolve output HTML file path by `manifest.json` file.
+    /// </summary>
+    private static string ResolveOutputHtmlRelativePath(string folder, string relativePath)
+    {
+        var manifestPath = Path.GetFullPath(Path.Combine(folder, "manifest.json"));
+        if (!File.Exists(manifestPath))
+            return string.Empty;
+
+        try
+        {
+            Manifest manifest = JsonUtility.Deserialize<Manifest>(manifestPath);
+
+            // Try to find output html file (html->html)
+            OutputFileInfo outputFileInfo = manifest.FindOutputFileInfo(relativePath);
+            if (outputFileInfo != null)
+                return outputFileInfo.RelativePath;
+
+            // Try to resolve output HTML file. (md->html)
+            relativePath = relativePath.Replace('\\', '/'); // Normalize path.
+            var manifestFile = manifest.Files
+                                       .Where(x => FilePathComparer.OSPlatformSensitiveRelativePathComparer.Equals(x.SourceRelativePath, relativePath))
+                                       .FirstOrDefault(x => x.OutputFiles.TryGetValue(".html", out outputFileInfo));
+
+            if (outputFileInfo != null)
+                return outputFileInfo.RelativePath;
+        }
+        catch (Exception)
+        {
+            throw; // Unexpected exception occurred.(e.g. Failed to deserialize Manifest)
+        }
+
+        // Failed to resolve output HTML file.
+        Logger.LogError($"Failed to resolve output HTML file path. sourceRelativePath: {relativePath}");
+        return string.Empty;
+    }
+
+    private static Process LaunchBrowser(string url)
+    {
+        try
+        {
+            // Windows
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return Process.Start("cmd", new[] { "/C", "start", url });
+            }
+
+            // Linux
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return Process.Start("xdg-open", url);
+            }
+
+            // OSX
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return Process.Start("open", url);
+            }
+
+            Logger.LogError($"Could not launch the browser process. Unknown OS platform: {RuntimeInformation.OSDescription}");
+        }
+        catch (Exception ex)
```

Comment:
nit: consider remove this try/catch block and leave it to the top-level exception handler.

Reply:
In latest commit.
Changed to fallback to return baseUrl.